pvlib/test/test_pvsystem.py

```diff
@@ -148,6 +148,87 @@ def test_PVSystem_physicaliam(mocker):
     assert iam < 1.
 
 
+def test_iam_martin_ruiz():
+
+    aoi = 45.
+    a_r = 0.16
+    expected = 0.98986965
+
+    # will fail of default values change
+    iam = pvsystem.iam_martin_ruiz(aoi)
+    assert_allclose(iam, expected)
+    # will fail of parameter names change
+    iam = pvsystem.iam_martin_ruiz(aoi=aoi, a_r=a_r)
+    assert_allclose(iam, expected)
+
+    a_r = 0.18
+    aoi = [-100, -60, 0, 60, 100, np.nan, np.inf]
+    expected = [0.0, 0.9414631, 1.0, 0.9414631, 0.0, np.nan, 0.0]
+
+    with np.errstate(invalid='ignore'):
```

Comment:
We usually try to handle numpy warnings in the function itself with either `with np.errstate` or masks. If I remember correctly, you were a proponent of putting `np.errstate` in more places in the library. Any particular reason you chose to put it here? I'm not necessarily opposed.

Reply:
Not sure why I did it that way; changed it it now.

Reply:
I do prefer the `np.errstate` approach over the masks.